cns/restserver/ipam.go

```diff
@@ -194,6 +194,26 @@ func (service *HTTPRestService) releaseIPConfig(podInfo cns.KubernetesPodInfo) e
 	return nil
 }
 
+// called when CNS is starting up and there are existing ipconfigs in the CRD that are marked as pending
+func (service *HTTPRestService) MarkExistingIPsAsPending(pendingIPIDs []string) error {
+	service.Lock()
+	defer service.Unlock()
+
+	for _, id := range pendingIPIDs {
+		if ipconfig, exists := service.PodIPConfigState[id]; exists {
+			if ipconfig.State == cns.Allocated {
+				return fmt.Errorf("Failed to mark IP [%v] as pending, currently allocated", id)
+			}
+
+			ipconfig.State = cns.PendingRelease
+			service.PodIPConfigState[id] = ipconfig
+		} else {
+			return fmt.Errorf("Inconsistent state, ipconfig with ID [%v] marked as pending release, but does not exist in state", id)
```

Comment:
Curious, would it be better to try to fix the spec at this point (i.e. garbage collect the ip id that is not referenced anywhere else)? Maybe that's done somewhere else but I'm not aware.

Reply:
This can be an expected state. say during reconcile, spec had some stale ToBeDeleted ips list which were already cleanup in status. Thus, PodIpConfigState wont have that ip. We can log this but this is not an error.